src/lib.rs

```diff
@@ -57,54 +57,108 @@ fn validate(payload: &[u8]) -> CallResult {
+        && !settings
+            .allowed_priority_classes
+            .contains(priority_class_name)
+    {
         return Err(format!(
             "Priority class \"{priority_class_name}\" is not allowed"
         ));
     }
-    Ok(())
+
+    Ok(POD_SPEC_NOT_MUTATE)
 }
 
 #[cfg(test)]
 mod tests {
     use super::validate_pod_priority_class;
+    use super::Settings;
+    use super::*;
     use rstest::rstest;
     use std::collections::HashSet;
 
     use k8s_openapi::api::core::v1::PodSpec;
 
     #[rstest]
-    #[case(Some("low-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), HashSet::new(), true)]
-    #[case(Some("no-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), false)]
-    #[case(Some("no-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), true)]
-    #[case(Some("low-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), false)]
-    #[case(None, HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), true)]
+    #[case(Some("low-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), HashSet::new(),None, true)]
+    #[case(Some("no-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), None, false)]
+    #[case(Some("no-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), None, true)]
+    #[case(Some("low-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), None, false)]
+    #[case(None, HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), None, true)]
+    // Tests with default priority class
+    #[case(None, HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), Some("low-priority".to_string()), true)]
+    #[case(None, HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),Some("no-priority".to_string()), true)]
+    #[case(Some("low-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), Some("no-priority".to_string()), true)]
+    #[case(Some("other-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),Some("no-priority".to_string()), true)]
+    #[case(Some("low-priority".to_string()), HashSet::new(), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]), Some("no-priority".to_string()), false)]
+    #[case(Some("no-priority".to_string()), HashSet::from(["high-priority".to_string(), "low-priority".to_string()]),HashSet::new(), Some("no-priority".to_string()), false)]
```

Comment:
I know it's not the fault of this PR, also it's a matter of personal taste, hence feel free to ignore this comment.

I would rewrite the signature of the test to be:

```rust
#[case] pod_priority_class: Option<&str>,
        #[case] allowed_classes: &[&str],
        #[case] denied_classes: &[&str],
        #[case] default_priority_class: Option<&str>,
        #[case] should_succeed: bool,
}
```

In this way the test cases definition would be more compact, hence more readable.

I would do all the `.to_string()` and `HashSet::from` into the body of the test.